packages/ipfs-core/src/components/repo/gc.js

```diff
@@ -17,6 +17,7 @@ const BLOCK_RM_CONCURRENCY = 256
  * @typedef {import('ipfs-core-types/src/refs').API} RefsAPI
  * @typedef {import('ipfs-repo')} IPFSRepo
  * @typedef {import('interface-datastore').Key} Key
+ * @typedef {import('ipfs-repo/src/types').Block} Block
```

Comment:
I had to make that reduced type in the interface over in `js-ipfs-repo` to resolve the other type error with Bitswap

Reply:
@achingbrain I'm actually gonna need a bit of assistance from you to verify the proper solution here.

The reason this is no ipld-block is I pushed a new commit to my branch on `js-ipfs-repo` to narrow the Block used for the Blockstore interface - https://github.com/hannahhoward/js-ipfs-repo/commit/dd812a61967dde6bd770df9a74bdc83565b97fe5

The reason I did this is cause of a type error that was happening in [ipfs-core/src/components/network.js](https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-core/src/components/network.js#L74) --- and here the reason is that IPFSBitswap expects a Blockstore that is defined in `ipfs-core-types`, which in turn uses a much more limited block interface.

Here's where I get lost -- `ipfs-core-types` is part of this repo, but looks like it was massively rewritten just recently? However, `ipfs-bitswap` seems to just reference the last release, which has this limited Blockstore interface -- it appears to be gone from the source code that's in this repo.

I feel like I lack the context here to understand what the proper typing solution is. What I have compiles and typechecks, but I think perhaps getting to ipld-block properly is something that should be handled by someone who knows what's going on with `ipfs-core-types`

Reply:
It's a bit in flux at the moment.  The initial ipfs-core-types was published in an earlier attempt at typing the API and contained a mishmash of shared interfaces, internal and external, as well as partial typings of other modules which should really live in those modules.  ipfs-bitswap used types from this module which on balance was a mistake as it creates a dependency loop that is hard to break.

Since then the various low-level modules have had types added, ipfs-core-types has been rewritten to only contain the public API and all the partial typings have been removed, so internals like `ipfs-bitswap` should not depend on it, instead getting types directly from the modules they use, unless they consume the public API, which they shouldn't as we're back to dependency loops again.

It's possible that bitswap and the repo (which were typed early in this exercise) need a pass to remove their own partial typings of other modules and other extraneous cruft.